tests/k8s_otel/kind_wrapper.py

```diff
@@ -0,0 +1,215 @@
+import time
+import random
+import subprocess
+import socket
+import docker
+import os
+from docker import DockerClient
+from pytest_kind import KindCluster
+from subprocess import CalledProcessError
+from contextlib import contextmanager
+from typing import Generator
+
+from jina.logging.logger import JinaLogger
+
+
+class KindClusterWrapperV2:
```

Comment:
I think this is overkill. For tests it might be easier to just run kubectl commands for deploying the otel components. The Python code adds overhead in terms of maintenance. You created a separate step anyway so it's isolated from the other k8s test and kind clustes.

Reply:
I agree with you actually. The python code here does just use kubectl in a subprocess in the end so it's not too different. New kubernetes API versions would not require a change in the python code since it just uses kubectl which should be updated instead. 

The old version of this wrapper in the other k8s test uses the python kubernetes client which could have this overhead of requiring an update separate from kubectl.

Oh and we can't just deploy otel components in a bash script outside of python because the lifecycle of the cluster is tied to python. Also exception handling and logging will ultimately need to be in pytest. So the python code does need to be the orchestrator in this case.

Reply:
I did try to do it without the wrapper. But it makes you duplicate a lot of common routines like loading images and apply then wait deployment. So the wrapper is probably a necessary evil. But I think this wrapper is less evil than the old one. The old one even pulls in linkerd though it doesn't seem like that is necessary and just creates an extra maintenance overhead of having to deal with linkerd versions